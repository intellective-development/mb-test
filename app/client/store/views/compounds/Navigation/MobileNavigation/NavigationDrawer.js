// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import { ui_actions } from 'store/business/ui';
import { user_selectors } from 'store/business/user';
import type { User } from 'store/business/user';

import DeliveryInfo from '../shared/DeliveryInfo';
import { MBAppStoreLink, MBDrawer, MBDynamicIcon, MBLink, MBText, MBTouchable } from '../../../elements';
import styles from './NavigationDrawer.scss';

type NavigationDrawerProps = {
  show: boolean,
  closeDrawer: () => void,

  // Connect Props
  user: User,
  showDeliveryInfoModal: () => void,
  showHelpModal: () => void
};
const NavigationDrawer = ({show, closeDrawer, user, showDeliveryInfoModal, showHelpModal}: NavigationDrawerProps) => {
  return (
    <MBDrawer
      open={show}
      closeDrawer={closeDrawer}
      className={styles.cmMNavDrawer_Container}>
      <div className={styles.cmMNavDrawer_Content}>
        <CloseLink handleClick={closeDrawer} />
        <DeliveryLink showDeliveryInfoModal={showDeliveryInfoModal} closeDrawer={closeDrawer} />
        <NavigationLinks user={user} closeDrawer={closeDrawer} showHelpModal={showHelpModal} />
        <MBAppStoreLink className={styles.cmMNavDrawer_AppStoreLink} />
      </div>
    </MBDrawer>
  );
};

const CloseLink = ({handleClick}) => (
  <MBTouchable onClick={handleClick}>
    <MBDynamicIcon name="x_close" width={30} height={30} className={styles.cmMNavDrawer_CloseLink} />
  </MBTouchable>
);

const DeliveryLink = ({showDeliveryInfoModal, closeDrawer}) => (
  <MBTouchable
    role="button"
    className={styles.cmMNavDrawer_DeliveryLink}
    onClick={() => {
      showDeliveryInfoModal();
      closeDrawer();
    }} >
    <DeliveryInfo />
    <MBDynamicIcon name="chevron_right" width={28} height={28} />
  </MBTouchable>
);

const NavigationLinks = ({user, closeDrawer, showHelpModal}) => (
  <ul className={styles.cmMNavDrawer_ItemLinkContainer}>
    <li>
      <MBLink.Text
        href="/store/"
        beforeNavigate={closeDrawer}
        className={styles.cmMNavDrawer_ItemLink}
        standard={false}>
        Shop
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/referrals"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Share & Save $10
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/account/orders"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Orders
      </MBLink.Text>
    </li>
    <li>
      <MBLink.Text
        href="/account/overview"
        className={styles.cmMNavDrawer_ItemLink}
        native_behavior
        standard={false}>
        Account
        <MBText.Span className={styles.cmMNavDrawer_UserName}>
          &emsp;{user && `${user.first_name} ${user.last_name.slice(0, 1)}.`}
        </MBText.Span>
      </MBLink.Text>
    </li>
    <li>
      <MBTouchable
        className={styles.cmMNavDrawer_ItemLink}
        onClick={() => {
          closeDrawer();
          showHelpModal();
        }} >
        <MBText.Span>Help</MBText.Span>
      </MBTouchable>
    </li>
  </ul>
);

const NavigationDrawerSTP = () => {
  const findUser = Ent.find('user');

  return (state) => ({user: findUser(state, user_selectors.currentUserId(state))});
};
const NavigationDrawerDTP = {
  showDeliveryInfoModal: ui_actions.showDeliveryInfoModal,
  showHelpModal: ui_actions.showHelpModal
};
const NavigationDrawerContainer = connect(NavigationDrawerSTP, NavigationDrawerDTP)(NavigationDrawer);

export default NavigationDrawerContainer;
