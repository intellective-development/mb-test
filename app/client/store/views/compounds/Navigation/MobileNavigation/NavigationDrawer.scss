@import "~client/store/views/style/variables";

.cmMNavDrawer_Container {
  background-color: $brand-white;
}

.cmMNavDrawer_Content {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.cmMNavDrawer_CloseLink {
  flex: none;
  padding: 9px 12px;
}

.cmMNavDrawer_DeliveryLink {
  flex: none;
  padding: 20px 6px 20px 4px; // icon (leftmost element) already has padding on it
  background-color: $brand-pale-blue;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.cmMNavDrawer_ItemLinkContainer {
  flex: auto;
  padding-top: 8px;
  margin: 0;
}

.cmMNavDrawer_ItemLink {
  @include font-size(24);
  font-weight: $medium-font-weight;
  padding: 14px 16px;
  display: inline-block;
}

.cmMNavDrawer_UserName {
  @include font-size(14);
  color: $brand-dark-grey;
}

.cmMNavDrawer_AppStoreLink {
  align-self: stretch;
  flex: none;
  padding: 0 $standard-gutter $standard-gutter;
  justify-content: space-around;
}
