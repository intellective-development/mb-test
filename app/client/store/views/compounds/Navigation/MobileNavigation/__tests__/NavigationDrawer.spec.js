
import React from 'react';
import address_factory from 'store/business/address/__tests__/address.factory';
import user_factory from 'store/business/user/__tests__/user.factory';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import NavigationDrawer from '../NavigationDrawer';

describe('NavigationDrawer', () => {
  it('renders', () => {
    const initial_state = {
      ...address_factory.stateify(
        address_factory.build({local_id: '10'})
      ),
      ...user_factory.stateify(
        user_factory.build({id: 1})
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <NavigationDrawer show />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders without user', () => {
    const initial_state = {
      ...address_factory.stateify(
        address_factory.build({local_id: '10'})
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <NavigationDrawer show />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders without address', () => {
    const initial_state = {
      ...user_factory.stateify(
        user_factory.build({id: 1})
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <NavigationDrawer show />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders a closed state', () => {
    expect(render(
      <TestProvider>
        <NavigationDrawer show={false} />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
