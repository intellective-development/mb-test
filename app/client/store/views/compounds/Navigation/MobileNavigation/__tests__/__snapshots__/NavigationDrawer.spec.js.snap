// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationDrawer renders 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay"
  />
  <div
    class="elMBDrawer"
  >
    <div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="30"
            viewBox="0 0 32 32"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.62 16.251L7.656 9.289 9.07 7.875l6.962 6.962 6.963-6.962 1.414 1.414-6.962 6.962 6.962 6.962-1.414 1.414-6.963-6.962-6.962 6.962-1.414-1.414 6.962-6.962z"
              fill="#222"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div>
          <img
            alt="on_demand"
            class="cmDeliveryMethodIcon_Multi"
            height="24"
            src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
            width="24"
          />
          <div>
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Delivery to
            </span>
            <br />
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               123 Jump Street, Apt. 10
            </span>
          </div>
        </div>
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="28"
            viewBox="0 0 32 32"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.297 10.922l.781-.625L19.64 16l-4.562 5.703-.78-.625L18.36 16z"
              fill="#494849"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <ul>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/store/"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/referrals"
          >
            Share & Save $10
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/orders"
          >
            Orders
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/overview"
          >
            Account
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               Brian C.
            </span>
          </a>
        </li>
        <li>
          <div
            class="el-touchable"
            role="button"
            tabindex="-1"
          >
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Help
            </span>
          </div>
        </li>
      </ul>
      <div
        class="elMBAppStoreLink_Container"
      >
        <a
          href="https://itunes.apple.com/us/app/minibar-delivery-wine-liquor/id720850888?mt=8&uo=4&at=11l5XB&ct=minibar"
          id="app-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Download on the App Store"
            src="/assets/components/elements/mb-app-store-link/ios-app-store.png"
            srcset="/assets/components/elements/mb-app-store-link/ios-app-store@2x.png 2x, /assets/components/elements/mb-app-store-link/ios-app-store@3x.png 3x"
          />
        </a>
        <a
          href="https://play.google.com/store/apps/details?id=minibar.android"
          id="play-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Get it on Google Play"
            src="/assets/components/elements/mb-app-store-link/play-store.png"
            srcset="/assets/components/elements/mb-app-store-link/play-store@2x.png 2x, /assets/components/elements/mb-app-store-link/play-store@3x.png 3x"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationDrawer renders a closed state 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay elMBDrawer_Overlay__Hidden"
  />
  <div
    class="elMBDrawer elMBDrawer__Hidden"
  >
    <div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="30"
            viewBox="0 0 32 32"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.62 16.251L7.656 9.289 9.07 7.875l6.962 6.962 6.963-6.962 1.414 1.414-6.962 6.962 6.962 6.962-1.414 1.414-6.963-6.962-6.962 6.962-1.414-1.414 6.962-6.962z"
              fill="#222"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div>
          <img
            alt="on_demand"
            class="cmDeliveryMethodIcon_Multi"
            height="24"
            src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
            width="24"
          />
          <div>
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Delivery to
            </span>
            <br />
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Add delivery address
            </span>
          </div>
        </div>
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="28"
            viewBox="0 0 32 32"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.297 10.922l.781-.625L19.64 16l-4.562 5.703-.78-.625L18.36 16z"
              fill="#494849"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <ul>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/store/"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/referrals"
          >
            Share & Save $10
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/orders"
          >
            Orders
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/overview"
          >
            Account
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               
            </span>
          </a>
        </li>
        <li>
          <div
            class="el-touchable"
            role="button"
            tabindex="-1"
          >
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Help
            </span>
          </div>
        </li>
      </ul>
      <div
        class="elMBAppStoreLink_Container"
      >
        <a
          href="https://itunes.apple.com/us/app/minibar-delivery-wine-liquor/id720850888?mt=8&uo=4&at=11l5XB&ct=minibar"
          id="app-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Download on the App Store"
            src="/assets/components/elements/mb-app-store-link/ios-app-store.png"
            srcset="/assets/components/elements/mb-app-store-link/ios-app-store@2x.png 2x, /assets/components/elements/mb-app-store-link/ios-app-store@3x.png 3x"
          />
        </a>
        <a
          href="https://play.google.com/store/apps/details?id=minibar.android"
          id="play-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Get it on Google Play"
            src="/assets/components/elements/mb-app-store-link/play-store.png"
            srcset="/assets/components/elements/mb-app-store-link/play-store@2x.png 2x, /assets/components/elements/mb-app-store-link/play-store@3x.png 3x"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationDrawer renders without address 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay"
  />
  <div
    class="elMBDrawer"
  >
    <div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="30"
            viewBox="0 0 32 32"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.62 16.251L7.656 9.289 9.07 7.875l6.962 6.962 6.963-6.962 1.414 1.414-6.962 6.962 6.962 6.962-1.414 1.414-6.963-6.962-6.962 6.962-1.414-1.414 6.962-6.962z"
              fill="#222"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div>
          <img
            alt="on_demand"
            class="cmDeliveryMethodIcon_Multi"
            height="24"
            src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
            width="24"
          />
          <div>
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Delivery to
            </span>
            <br />
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Add delivery address
            </span>
          </div>
        </div>
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="28"
            viewBox="0 0 32 32"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.297 10.922l.781-.625L19.64 16l-4.562 5.703-.78-.625L18.36 16z"
              fill="#494849"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <ul>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/store/"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/referrals"
          >
            Share & Save $10
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/orders"
          >
            Orders
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/overview"
          >
            Account
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               Brian C.
            </span>
          </a>
        </li>
        <li>
          <div
            class="el-touchable"
            role="button"
            tabindex="-1"
          >
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Help
            </span>
          </div>
        </li>
      </ul>
      <div
        class="elMBAppStoreLink_Container"
      >
        <a
          href="https://itunes.apple.com/us/app/minibar-delivery-wine-liquor/id720850888?mt=8&uo=4&at=11l5XB&ct=minibar"
          id="app-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Download on the App Store"
            src="/assets/components/elements/mb-app-store-link/ios-app-store.png"
            srcset="/assets/components/elements/mb-app-store-link/ios-app-store@2x.png 2x, /assets/components/elements/mb-app-store-link/ios-app-store@3x.png 3x"
          />
        </a>
        <a
          href="https://play.google.com/store/apps/details?id=minibar.android"
          id="play-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Get it on Google Play"
            src="/assets/components/elements/mb-app-store-link/play-store.png"
            srcset="/assets/components/elements/mb-app-store-link/play-store@2x.png 2x, /assets/components/elements/mb-app-store-link/play-store@3x.png 3x"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavigationDrawer renders without user 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay"
  />
  <div
    class="elMBDrawer"
  >
    <div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="30"
            viewBox="0 0 32 32"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.62 16.251L7.656 9.289 9.07 7.875l6.962 6.962 6.963-6.962 1.414 1.414-6.962 6.962 6.962 6.962-1.414 1.414-6.963-6.962-6.962 6.962-1.414-1.414 6.962-6.962z"
              fill="#222"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <div
        class="el-touchable"
        role="button"
        tabindex="-1"
      >
        <div>
          <img
            alt="on_demand"
            class="cmDeliveryMethodIcon_Multi"
            height="24"
            src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
            width="24"
          />
          <div>
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Delivery to
            </span>
            <br />
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               123 Jump Street, Apt. 10
            </span>
          </div>
        </div>
        <div
          class="elMBDynamicIcon"
        >
          <svg
            class=""
            height="28"
            viewBox="0 0 32 32"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.297 10.922l.781-.625L19.64 16l-4.562 5.703-.78-.625L18.36 16z"
              fill="#494849"
              fill-rule="nonzero"
            />
          </svg>
        </div>
      </div>
      <ul>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/store/"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/referrals"
          >
            Share & Save $10
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/orders"
          >
            Orders
          </a>
        </li>
        <li>
          <a
            class="elMBText_A elMBText elMBText_ResetSpacing"
            href="/account/overview"
          >
            Account
            <span
              class="elMBText elMBText_ResetSpacing"
            >
               
            </span>
          </a>
        </li>
        <li>
          <div
            class="el-touchable"
            role="button"
            tabindex="-1"
          >
            <span
              class="elMBText elMBText_ResetSpacing"
            >
              Help
            </span>
          </div>
        </li>
      </ul>
      <div
        class="elMBAppStoreLink_Container"
      >
        <a
          href="https://itunes.apple.com/us/app/minibar-delivery-wine-liquor/id720850888?mt=8&uo=4&at=11l5XB&ct=minibar"
          id="app-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Download on the App Store"
            src="/assets/components/elements/mb-app-store-link/ios-app-store.png"
            srcset="/assets/components/elements/mb-app-store-link/ios-app-store@2x.png 2x, /assets/components/elements/mb-app-store-link/ios-app-store@3x.png 3x"
          />
        </a>
        <a
          href="https://play.google.com/store/apps/details?id=minibar.android"
          id="play-store-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Get it on Google Play"
            src="/assets/components/elements/mb-app-store-link/play-store.png"
            srcset="/assets/components/elements/mb-app-store-link/play-store@2x.png 2x, /assets/components/elements/mb-app-store-link/play-store@3x.png 3x"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
